src/ralph/backends/data/async_ws.py

```diff
@@ -0,0 +1,183 @@
+"""Websocket stream backend for Ralph."""
+
+from typing import AsyncIterator, Dict, Optional, Union
+
+import websockets
+from pydantic import PositiveInt
+
+from ralph.backends.data.base import (
+    BaseAsyncDataBackend,
+    BaseDataBackendSettings,
+    BaseQuery,
+    DataBackendStatus,
+)
+from ralph.conf import BaseSettingsConfig
+from ralph.exceptions import BackendException
+from ralph.utils import async_parse_iterable_to_dict
+
+
+class WSDataBackendSettings(BaseDataBackendSettings):
+    """Websocket data backend default configuration.
+
+    Attributes:
+        URI (str): The URI to connect to.
+    """
+
+    class Config(BaseSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__DATA__WS__"
+
+    URI: Optional[str] = None
+
+
+class AsyncWSDataBackend(BaseAsyncDataBackend[WSDataBackendSettings, BaseQuery]):
+    """Websocket stream backend."""
+
+    name = "async_ws"
+
+    def __init__(self, settings: Optional[WSDataBackendSettings] = None):
+        """Instantiate the async websocket data backend.
+
+        Args:
+            settings (WSDataBackendSettings or None): The websocket backend settings.
+                If `settings` is `None`, a default settings instance is used instead.
+        """
+        super().__init__(settings)
+        self._clients: Dict[str, websockets.WebSocketClientProtocol] = {}
```

Comment:
Why choosing to support multiple simultaneous connections instead of one per instance as it was previously implemented?

Reply:
Good point) The idea was to align with other data backends and use the `target` argument for the URI—the fallback to the `settings.URI ` was implemented in the body of the `_read_bytes` method. However, I agree with you; limiting one connection per instance is much easier and straightforward to handle, Updated it)